Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -27,6 +27,11 @@ public struct DeserializeMiddleware<OperationStackOutput>: Middleware {
 
             let response = try await next.handle(context: context, input: input) // call handler to get http response
 
+            if let responseDateString = response.httpResponse.headers.value(for: "Date") {
+                let estimatedSkew = getEstimatedSkew(now: Date(), responseDateString: responseDateString)
+                context.attributes.set(key: AttributeKeys.estimatedSkew, value: estimatedSkew)
+            }
```

Comment:
Calculate `estimatedSkew` by getting difference between server's timestamp on HTTP response returned via the `Date` header and current client time. Save to `context` for use in `RetryMiddleware`.

Reply:
What happens when the response body takes a long time to transmit due to size or bad network?  Does that time affect this number?

Reply:
Bad network would increase estimated skew because network latency is included in the estimated skew. 

As for response body taking a long time to transmit, that would increase estimated skew as well since that would delay when the response is deserialized which would subsequently delay when the estimated skew gets calculated.